### PR TITLE
fix: update gitignore with dist folder

### LINKED ISSUE #14607: TS .gitignore missing "dist" folder
## Bug report
hi just found that `.gitignore` missing `dist` folder, witch is appearing when TS project is build

### Required System information

- Strapi version: 4.4.3

### Steps to reproduce the behavior

1. Create strapi typescipt project
2. Buid project 
3. Notice 'dist' folder
4. Git commit.... 
5. Joy...

### PR DESCRIPTION
**Pull request by ilteoood**
### What does it do?

Closes #14607, updating the template `.gitignore` 

### Why is it needed?

In TS projects, `dist` folder is not ignored

### How to test it?

Create a strapi project with TypeScript

### Related issue(s)/PR(s)

#14607
cc: @simoneb

### CODE REVIEW

**Comment by ilteoood** [comment]
ping @derrickmehaffy @innerdvations

**Comment by innerdvations** [comment]
Thanks for the contribution!

From my initial check, it fixes `create-strapi-starter` but I noticed we still have the same problem for `create-strapi-app`. I'd like to fix it there (in `packages/generators/app/lib/resources/dot-files/common/gitignore` and also update the .gitignore files of the starter projects so they mirror what would be created.

If you'd like to make those changes to this PR, please go ahead, otherwise I can take care of it sometime next week.

**Comment by ilteoood** [comment]
Hi @innerdvations,
thank you so much for your suggestions, I applied all of them

File: packages/cli/create-strapi-starter/resources/gitignore [hunk 1]
```diff
@@ -111,4 +111,5 @@ license.txt
 exports
 *.cache
 build
+dist
```
Comment (innerdvations) [inline packages/cli/create-strapi-starter/resources/gitignore:114]:
```suggestion
dist
build
```

To keep everything in the same order

**Review by innerdvations — APPROVED** [review]

**Pull request merged**